conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DialogueServiceGenerator.java

```diff
@@ -69,11 +69,19 @@ public DialogueServiceGenerator(Options options) {
         TypeNameResolver typeNameResolver = typeName -> Preconditions.checkNotNull(
                 typeDefinitionsByName.get(typeName), "Referenced unknown TypeName", SafeArg.of("typeName", typeName));
 
-        AsyncGenerator asyncGenerator = new AsyncGenerator(
-                options, typeNameResolver, new ParameterTypeMapper(parameterTypes), new ReturnTypeMapper(returnTypes));
+        StaticFactoryMethodGenerator asyncGenerator = new DefaultStaticFactoryMethodGenerator(
+                options,
+                typeNameResolver,
+                new ParameterTypeMapper(parameterTypes),
+                new ReturnTypeMapper(returnTypes),
+                StaticFactoryMethodType.Async);
 
-        BlockingGenerator blockingGenerator = new BlockingGenerator(
-                options, new ParameterTypeMapper(parameterTypes), new ReturnTypeMapper(returnTypes));
+        StaticFactoryMethodGenerator blockingGenerator = new DefaultStaticFactoryMethodGenerator(
+                options,
+                typeNameResolver,
+                new ParameterTypeMapper(parameterTypes),
+                new ReturnTypeMapper(returnTypes),
```

Comment:
We could reuse the same ParameterTypeMapper and ReturnTypeMapper objects, probably makes no difference in practice though

Reply:
See my comment above, I think this whole thing should be made so that you instantiate one instance for each class we're generating, to simplify everything.